avro-serializer/src/main/java/io/confluent/kafka/serializers/KafkaAvroDeserializer.java

```diff
@@ -53,6 +54,11 @@ public Object deserialize(String s, byte[] bytes) {
     return deserialize(bytes);
   }
 
+  /**
+   * Pass a reader schema to get an Avro projection
+   */
+  public Object deserialize(String s, byte[] bytes, Schema readerSchema) { return deserialize(bytes, readerSchema); }
```

Comment:
The first version I wrote worked the way you described but I felt it was painful to required a new KafkaAvroDecoder object for each projection.  It makes configuration very awkward.  In my use case below, I need multiple projections of a given message and would be hard to create a configuration that sets up a different AvroSerde/KafkaAvroDecoder object for each one.  

Here's an example of how I use it in a Samza stream-table join job.  I've wrapped the Confluent decoder in a [Samza serde](https://github.com/quantiply/rico/blob/master/avro/src/main/java/com/quantiply/samza/serde/AvroSerde.java)

```
    //For new lookup messages, save them in the local KV store
    private void processDeployMsg(IncomingMessageEnvelope envelope, MessageCollector collector, TaskCoordinator coordinator) throws Exception {
        //Note that we're decoding the same binary message with the same serde/decoder
        //into two different projection objects, one for the primary key fields, one for the values.
        SvcAttributeValue attrValue = (SvcAttributeValue) avroSerde.fromBytes((byte[]) envelope.getMessage(), SvcAttributeValue.getClassSchema());
        SvcCallEventKey key = (SvcCallEventKey) avroSerde.fromBytes((byte[]) envelope.getMessage(), SvcCallEventKey.getClassSchema());
        //save in local KV store
        deployAttrs.put(key, attrValue);
    }

    //For new events, extract the key, lookup the join data and merge it
    private void processCallMsg(IncomingMessageEnvelope envelope, MessageCollector collector, TaskCoordinator coordinator, EventStreamMetrics streamMetrics) throws Exception {
        //Get all the data (not a projection)
        SvcCallEvent call = (SvcCallEvent) avroSerde.fromBytes((byte []) envelope.getMessage());
        //Get the lookup key by projection
        SvcCallEventKey key = (SvcCallEventKey) avroSerde.fromBytes((byte []) envelope.getMessage(), SvcCallEventKey.getClassSchema());
        SvcAttributeValue attr = deployAttrs.get(key);

        GenericData.Record enriched = new Join(SvcCallEventWDeploy.getClassSchema())
                .merge(call)
                .merge(attr)
                .getBuilder()
                .set("header", getCamusHeaders(call.getHeader(), Headers.getClassSchema(), tsNowMs))
                .build();

        collector.send(new OutgoingMessageEnvelope(outSysStream, avroSerde.toBytes(enriched)));
    }
```